android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java

```diff
@@ -265,41 +266,39 @@ public boolean isShowing() {
     private void setUpObservers() {
         cleanUpObservers();
         mWalletModel.getKeyringModel().mAllAccountsInfo.observeForever(mAllAccountsInfoObserver);
-        mWalletModel.getCryptoModel().getNetworkModel().mDefaultNetwork.observeForever(
-                mDefaultNetworkObserver);
+        getNetworkModel().mDefaultNetwork.observeForever(mDefaultNetworkObserver);
     }
 
     private void cleanUpObservers() {
         mWalletModel.getKeyringModel().mAllAccountsInfo.removeObserver(mAllAccountsInfoObserver);
-        mWalletModel.getCryptoModel().getNetworkModel().mDefaultNetwork.removeObserver(
-                mDefaultNetworkObserver);
+        getNetworkModel().mDefaultNetwork.removeObserver(mDefaultNetworkObserver);
     }
 
-    private void updateAccountInfo(AccountInfo selectedAccount) {
+    private void updateSelectedAccountInfo() {
         Utils.setBlockiesBitmapResource(
-                mExecutor, mHandler, mAccountImage, selectedAccount.address, true);
+                mExecutor, mHandler, mAccountImage, mSelectedAccount.address, true);
         Utils.setBlockiesBackground(
-                mExecutor, mHandler, mContainerConstraintLayout, selectedAccount.address, true);
-        mAccountName.setText(selectedAccount.name);
-        mAccountAddress.setText(Utils.stripAccountAddress(selectedAccount.address));
-        getBalance(selectedAccount);
-        updateAccountConnected(selectedAccount);
+                mExecutor, mHandler, mContainerConstraintLayout, mSelectedAccount.address, true);
+        mAccountName.setText(mSelectedAccount.name);
+        mAccountAddress.setText(Utils.stripAccountAddress(mSelectedAccount.address));
+        updateSelectedAccountBalance();
+        updateSelectedAccountConnectionState();
     }
 
-    private void updateAccountConnected(AccountInfo selectedAccount) {
-        if (CoinType.SOL == selectedAccount.accountId.coin) {
+    private void updateSelectedAccountConnectionState() {
+        if (CoinType.SOL == mSelectedAccount.accountId.coin) {
             if (mAccountsWithPermissions.size() == 0) {
                 mBtnConnectedStatus.setText("");
                 mBtnConnectedStatus.setVisibility(View.GONE);
                 mCvSolConnectionStatus.setVisibility(View.GONE);
             } else {
-                isSolanaConnected(selectedAccount);
+                isSolanaConnected(mSelectedAccount);
```

Comment:
nit: `isSolanaConnected` should be renamed to `updateSolanaConnected`, might have been missed during initial review.

Reply:
https://github.com/brave/brave-core/pull/19329/commits/2b62a3db430107196b574a0d05990d585d0ee963